src/Microsoft.ML.Core/Prediction/ITrainer.cs

```diff
@@ -27,151 +26,126 @@ namespace Microsoft.ML.Runtime
+    /// <summary>
+    /// Strongly typed generic interface for a trainer. A trainer object takes training data
+    /// and produces a predictor.
+    /// </summary>
+    /// <typeparam name="TPredictor"> Type of predictor produced</typeparam>
+    public interface ITrainer<out TPredictor> : ITrainer
         where TPredictor : IPredictor
     {
-        TPredictor CombineModels(IEnumerable<TModel> models);
+        /// <summary>
+        ///  Trains a predictor.
+        /// </summary>
+        /// <param name="context">A context containing at least the training data</param>
+        /// <returns>The trained predictor</returns>
+        new TPredictor Train(TrainContext context);
     }
 
-    public delegate void SignatureModelCombiner(PredictionKind kind);
-
-    /// <summary>
-    /// Weakly typed interface for a trainer "session" that produces a predictor.
-    /// </summary>
-    public interface ITrainer
+    public static class TrainerExtensions
     {
         /// <summary>
-        /// Return the type of prediction task for the produced predictor.
+        /// Convenience train extension for the case where one has only a training set with no auxiliary information.
+        /// Equivalent to calling <see cref="ITrainer.Train(TrainContext)"/>
+        /// on a <see cref="TrainContext"/> constructed with <paramref name="trainData"/>.
         /// </summary>
-        PredictionKind PredictionKind { get; }
+        /// <param name="trainer">The trainer</param>
+        /// <param name="trainData">The training data.</param>
+        /// <returns>The trained predictor</returns>
+        public static IPredictor Train(this ITrainer trainer, RoleMappedData trainData)
+            => trainer.Train(new TrainContext(trainData));
 
         /// <summary>
-        ///  Returns the trained predictor.
-        ///  REVIEW: Consider removing this.
+        /// Convenience train extension for the case where one has only a training set with no auxiliary information.
+        /// Equivalent to calling <see cref="ITrainer{TPredictor}.Train(TrainContext)"/>
+        /// on a <see cref="TrainContext"/> constructed with <paramref name="trainData"/>.
         /// </summary>
-        IPredictor CreatePredictor();
+        /// <param name="trainer">The trainer</param>
+        /// <param name="trainData">The training data.</param>
+        /// <returns>The trained predictor</returns>
+        public static TPredictor Train<TPredictor>(this ITrainer<TPredictor> trainer, RoleMappedData trainData) where TPredictor : IPredictor
+            => trainer.Train(new TrainContext(trainData));
     }
 
     /// <summary>
-    /// Interface implemented by the MetalinearLearners base class.
-    /// Used to distinguish the MetaLinear Learners from the other learners
+    /// Interface to provide extra information about a trainer.
     /// </summary>
-    public interface IMetaLinearTrainer
+    public interface ITrainerEx : ITrainer
```

Comment:
I worry about the future proofing of this interface.  What about exposing a capabilities class that could be extended?

Reply:
Huh... actually I like that idea a lot.

Reply:
Yes that was what I was thinking.  Bonus that you can get rid of Ex.  Seems like it can be reasonably lightweight to implement.

Reply:
Great. If though it contains information about whether it needs normalization, or desires caching, or whatever, that's less about capabilities. Hmmm, I don't like how generic the name is, but would `TrainerInfo` be that bad of a name, compared to `TrainerCapabilities`? I don't like how generic the name info is though... *everything* is info. 😛 Anyway I'll just whip something up, we can always change the name.